rosbridge_library/src/rosbridge_library/protocol.py

```diff
@@ -109,16 +114,27 @@ def __init__(self, client_id: str, node_handle: Node) -> None:
         :param node_handle: A ROS2 node handle
         """
         self.client_id = client_id
+        self.node_handle = node_handle
+        self.parameters = parameters
+
         self.capabilities: list[Capability] = []
         self.operations: dict[str, Callable[[dict[str, Any]], None]] = {}
-        self.node_handle = node_handle
         self.external_service_list = {}
         self.external_action_list = {}
 
         if self.parameters:
-            self.fragment_size = self.parameters["max_message_size"]
-            self.delay_between_messages = self.parameters["delay_between_messages"]
-            self.bson_only_mode = self.parameters.get("bson_only_mode", False)
+            if "max_message_size" in self.parameters:
```

Comment:
You can probably do repetitive stuff like this in a for loop over all the parameter names

Reply:
Something like that?
```
        if self.parameters:
            for param_name in ("max_message_size", "delay_between_messages", "bson_only_mode"):
                if param_name in self.parameters:
                    setattr(self, param_name, self.parameters[param_name])
```